src/s3/commands/uploadFile.ts

```diff
@@ -26,51 +38,103 @@ export interface FileSizeBytes {
 }
 
 /**
- * Uploads a file to the bucket or folder represented by the given node.
- *
- * Prompts the user for the file location.
- * Shows the output channel with "upload started" message.
- * Uploads the file (showing a progress bar).
- * Shows the output channel with "upload completed" message.
- * Refreshes the node.
- *
- * Node that the node is reset to displaying its first page of results.
- * The file that is uploaded won't necessary fall on the first page.
- * The user may need to load more pages to see the uploaded file reflected in the tree.
+ * Wizard to upload a file.
+ * If the S3BucketNode | S3FolderNode is given, it asks for the file then uploads to node
+ * If the node is not give, it asks for the file and also for a bucket, then uploads
+ * 
+ * @param s3Client - account to upload the file to
+ * @param node - bucket or folder to upload the file to
+ * @param document - open document (to use as default), if any
+ * 
  */
-export async function uploadFileCommand(
-    node: S3BucketNode | S3FolderNode,
+ export async function uploadFileCommand(
+    s3Client: S3Client,
+    node?: S3BucketNode | S3FolderNode,
+    document?: vscode.Uri,
```

Comment:
`node` and `document` are mutually-exclusive parameters. Only one can be given. This could be guarded with `assert()`, but instead I suggest collapsing to a single parameter:

     /** @param nodeOrDocument - bucket/folder node to upload to, OR file URI to be uploaded */
     nodeOrDocument?: S3BucketNode | S3FolderNode | vscode.Uri

Reply:
Good suggestion